src/marqo/tensor_search/add_docs.py

```diff
@@ -20,8 +22,10 @@ def threaded_download_images(allocated_docs: List[dict], image_repo: dict,
         allocated_docs: docs with images to be downloaded by this thread,
         image_repo: dictionary that will be mutated by this thread. It will add PIL images
             as values and the URLs as keys
+        tensor_fields: A tuple of tensor_fields. Images will be downloaded for these fields only. Cannot be provided
+            at the same time as `non_tensor_fields`.
         non_tensor_fields: A tuple of non_tensor_fields. No images will be downloaded for
-            these fields
+            these fields. Cannot be provided at the same time as `tensor_fields`.
```

Comment:
We don't validate that `non_tensor_fields ` and `tensor_fields` aren't allowed to co-occur in this function. We should:
a) if we expected validation to be done beforehand we should explicitly state this in the docstring, along with the function/code that is expected to do this validation. This is to inform future devs in case they want to refactor
b) validate it here, and raise an internal error because this is an unusual state. If this is an unusual state because a user error should have been raised earlier, this should be stated in the docstring

Reply:
Addressed